force-app/main/default/classes/SOQL.cls

```diff
@@ -85,20 +121,46 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+        
+    public static Map<Id, Task> getTasksDueInNextSevenDays() {
+        // Query for tasks due within the next 7 days
+        Date nextWeek = Date.today().addDays(7);
+        Date today = Date.today();
+    
+        // Create a map to store Task Ids to Task records
+        Map<Id, Task> taskMap = new Map<Id, Task>();
+    
+        // Query for tasks due within the next 7 days
+        for(Task t : [
+            SELECT Id, Subject, ActivityDate 
+            FROM Task 
+            WHERE ActivityDate >= :today 
+            AND ActivityDate <= :nextWeek
+            ]) {
+            taskMap.put(t.Id, t);
+        }
```

Comment:
Great SOQL for loop. Exactly the use case for it.

There are some key words you can use in queries to do date functions, such as TODAY, YESTERDAY, LAST_N_DAYS:n, and others. Sometimes those can enable you to avoid instantiating Date or Datetime variables to compare against.